musl does not yet have the EF_RISCV_* constants in elf.h. Poly/ML only
uses a single constant from this file, hence we can just hardcode it.

diff -upr polyml-5.9.1.orig/libpolyml/elfexport.cpp polyml-5.9.1/libpolyml/elfexport.cpp
--- polyml-5.9.1.orig/libpolyml/elfexport.cpp	2023-09-21 12:57:37.455768326 +0200
+++ polyml-5.9.1/libpolyml/elfexport.cpp	2023-09-21 12:58:50.852658976 +0200
@@ -57,6 +57,13 @@
 #include <elf_abi.h>
 #endif
 
+/* Contrary to glibc, musl does not define this constant
+ * in elf.h hence we just copy the definition from binutil's
+ * include/elf/riscv.h file here manually. */
+#ifndef EF_RISCV_FLOAT_ABI_DOUBLE
+#define EF_RISCV_FLOAT_ABI_DOUBLE 0x0004
+#endif
+
 #ifdef HAVE_MACHINE_RELOC_H
 #include <machine/reloc.h>
 
